gst-plugin/src/drpai-models/base_post_processor.cpp

```diff
@@ -0,0 +1,115 @@
+//
+// Created by matin on 03/11/24.
+//
+
+#include "base_post_processor.h"
+#include <fstream>
+
+/// Renders bounding boxes on the image using the latest detections.
+/// This function usually gets overridden by the child class. But it is not a must.
+/// @param [in] img Reference to the image to be rendered on.
+void BasePostProcessor::render_detections_on_image(Image &img) {
+    std::unique_lock lock (mutex);
+    for (const auto& detection: last_det)
+    {
+        /* Draw the bounding box on the image */
+        img.draw_rect(detection.bbox, detection.to_string_hr());
+    }
+}
+
+/// Renders texts at the corner of the image using the list of corner texts.
+/// This function usually gets overridden by the child class. But it is not a must.
+/// @param [in] img Reference to the image to be rendered on.
+void BasePostProcessor::render_text_on_image(Image &img) {
+    for(std::size_t i=0; i<corner_text.size(); i++) {
+        img.write_string(corner_text.at(i), 0, static_cast<int32_t>(i*15), WHITE_DATA, BLACK_DATA, 5);
+    }
+}
+
+/// Get a json array containing detections to be used in the get_json function.
+/// This function can get overridden by the child class to add extra info for each detection.
+/// @returns A json_array containing detections
```

Comment:
What carries the detections and who provides them?

Reply:
Right. I will give more explanation.